File: FragmentExample/app/src/main/java/com/example/fragmentexample/FirstFragment.java
package com.example.fragmentexample;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;


public class FirstFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        Toast.makeText(getContext(),"onCreateView",Toast.LENGTH_SHORT).show();
        return inflater.inflate(R.layout.fragment_first, container, false);
    }

    @Override
    public void onStart() {
        super.onStart();
        Toast.makeText(getContext(),"onStart",Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onResume() {
        super.onResume();
        Toast.makeText(getContext(),"onResume",Toast.LENGTH_SHORT).show();

    }

    @Override
    public void onPause() {
        super.onPause();
        Toast.makeText(getContext(),"onPause",Toast.LENGTH_SHORT).show();

    }

    @Override
    public void onStop() {
        super.onStop();
        Toast.makeText(getContext(),"onStop",Toast.LENGTH_SHORT).show();

    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Toast.makeText(getContext(),"onDestroy",Toast.LENGTH_SHORT).show();

    }
}

File: NavigationComponent/app/src/main/java/com/example/navigationcomponent/FirstFragment.kt
package com.example.navigationcomponent

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import kotlinx.android.synthetic.main.fragment_first.*

class FirstFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_first, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val controller = findNavController()

        btn_open_fragment_two.setOnClickListener {
            /* val bundle = bundleOf(
                 "userName" to edt_name.text.toString()
             )*/
            val user = User(edt_name.text.toString(), edt_age.text.toString().toInt())
            /*  val bundle= bundleOf(
                  "user" to user
              )*/
            val address = "Ha Noi"

            val action = FirstFragmentDirections.actionFirstFragmentToSecondFragment(address, user)

            controller.navigate(action)
        }

        btn_open_fragment_three.setOnClickListener {
            controller.navigate(R.id.thirdFragment)
        }

    }


}